modules/widgets/contact-info.php

```diff
@@ -242,11 +242,37 @@ function form( $instance ) {
 			<p>
 				<label for="<?php echo esc_attr( $this->get_field_id( 'address' ) ); ?>"><?php esc_html_e( 'Address:', 'jetpack' ); ?></label>
 				<textarea class="widefat" id="<?php echo esc_attr( $this->get_field_id( 'address' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'address' ) ); ?>"><?php echo esc_textarea( $instance['address'] ); ?></textarea>
+			</p>
 
+			<p>
 				<input class="jp-contact-info-showmap" id="<?php echo esc_attr( $this->get_field_id( 'showmap' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'showmap' ) ); ?>" value="1" type="checkbox" <?php checked( $instance['showmap'], 1 ); ?> />
 				<label for="<?php echo esc_attr( $this->get_field_id( 'showmap' ) ); ?>"><?php esc_html_e( 'Show map', 'jetpack' ); ?></label>
 			</p>
 
+			<?php if ( has_filter( 'jetpack_google_maps_api_key' ) ) { ?>
+
+			<p class="jp-contact-info-admin-map" style="<?php echo $instance['showmap'] ? '' : 'display: none;'; ?>">
+				<?php
+					printf(
+						wp_kses(
+							__('API key <code>%1$s</code> is being set by <a href="%2$s" target="_blank" rel="noopener noreferrer">this</a> filter.', 'jetpack'),
```

Comment:
Personally, I'd much rather not display anything, not even a message about the filter. I think that one of the goals of using the filter for a site owner could be to declutter the widget settings area and make it less confusing for other admins. As a result, removing all the API key UI would make sense in such cases imo. I would only leave the "Show Map" checkbox.

One good use-case (and probably the most used out there) of that filter is Atomic, or WordPress.com, for example. I don't think it will be valuable for WordPress.com and Atomic site owners to know about the filter.

Reply:
Good points! I agree with all of them.

My concern was that users might have trouble fixing any API key problems if they don’t know that the API key is required and that it was set by another user with the filter. But, I think that situation will be very rare and can be addressed with the support documentation. The vast majority of users will probably use the widget’s API key input.